pipeline_steps/common.groovy

```diff
@@ -439,4 +439,48 @@ def is_doc_update_pr(git_dir) {
   return is_doc_update_pr
 }
 
+/* Look for JIRA issue key in commit messages for commits in the source branch
+ * that aren't in the target branch.
+ * This function uses environment variables injected by github pull request
+ * builder and so can only be used for PR triggered jobs
+ */
+def get_jira_issue_key(repo_path="rpc-openstack"){
+  def key_regex = "^[a-zA-Z][a-zA-Z0-9_]+-[1-9][0-9]*"
+  dir(repo_path){
+    commit_titles = sh(
+      returnStdout: true,
+      script: "git log --pretty=%s origin/${ghprbTargetBranch}..${ghprbSourceBranch}").split('\n')
+    print("looking for Jira issue keys in the following commits: ${commit_titles}")
+    for (def i=0; i<=commit_titles.size(); i++){
+      try{
+        key = (commit_titles[i] =~ key_regex)[0]
+        print ("Found Jira Issue Key: ${key}")
+        return key
+      } catch (e){
+        continue
+      }
+    }
+    throw new Exception("JIRA Issue key not found in any of the commits"
+                        +" introduced by ${repo_path}:${ghprbSourceBranch}")
+  }
+}
+
+/* Attempt to add a jira comment, but don't fail if ghprb env vars are missing
+ * or no Jira issue key is present in commit titles
+ */
+def safe_jira_comment(body, repo_path="rpc-openstack"){
+  if (env.ghprbTargetBranch == null){
+    print ("Not a PR job, so not attempting to add a Jira comment")
+    return
+  }
+  try{
+    key = get_jira_issue_key(repo_path)
+    jiraComment(issueKey: key,
+                body: body)
+    print "Jira Comment: [${key}] ${body}"
+  } catch (e){
+    print (e)
```

Comment:
It might be useful to prefix the error string with a standard bit of blurb to explain, in general, what is failing.